### PR TITLE
SOF-7208: add shallowDeepAlmostEqual

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

File: src/js/shared/compare.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+export function shallowDeepAlmostEqual(
```
Comment (timurbazhirov) [inline src/js/shared/compare.ts]: maybe assert.js file and assert in the name?

File: dist/js/shared/compare.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+"use strict";
```
Comment (timurbazhirov) [inline dist/js/shared/compare.js]: Needs to be cleaned

File: src/js/shared/compare.ts (hunk 1 above)
Comment (k0stik) [reply src/js/shared/compare.ts]: Done

File: dist/js/shared/compare.js (hunk 2 above)
Comment (k0stik) [reply dist/js/shared/compare.js]: Done

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**